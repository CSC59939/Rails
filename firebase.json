{
  "functions": {
    "predeploy": [
      "npm --prefix \"$RESOURCE_DIR\" run lint"
    ]
  },
  "hosting": {
    "public": "website/build",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ]
  }
}
