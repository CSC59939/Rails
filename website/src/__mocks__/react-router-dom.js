import React from 'react';
const rrd = require('react-router-dom');
rrd.BrowserRouter = ({children}) => <div>{children}</div>
module.exports = rrd;
// this is a manual mock of Jest, which it redefine the browserRouter, since it will conflict with memory router.
