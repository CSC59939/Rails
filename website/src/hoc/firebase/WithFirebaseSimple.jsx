import React, { PureComponent } from 'react';
import firebase from 'firebase/app';
import 'firebase/auth';

function WithFirebaseSimple(App) {
  return class extends PureComponent {
    render() {
      return (
        <App
          {...this.props}
          firebase={firebase}
        />
      );
    }
  };
}

export default WithFirebaseSimple;
