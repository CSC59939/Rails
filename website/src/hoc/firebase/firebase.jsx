import React, { PureComponent } from 'react';
import firebase from 'firebase/app';
import 'firebase/auth';

function withFirebase(App) {
  return class extends PureComponent {
    constructor(props) {
      super(props);
      this.state = {
        signedin: false,
      };
      this.firebaseListener = null;
      this.authStateListener = this.authStateListener.bind(this);
      this.signoutHandler = this.signoutHandler.bind(this);
      if (firebase.app()) this.authStateListener();
    }

    componentDidMount() {
      // if (firebase.app()) this.authStateListener();
    }

    componentWillUnmount() {
      if (this.firebaseListener) {
        this.firebaseListener();
      }
      this.authStateListener = undefined;
    }

    authStateListener() {
      this.firebaseListener = firebase.auth().onAuthStateChanged((user) => {
        if (user) {
          this.setState({ signedin: true });
        } else {
          this.setState({ signedin: false });
        }
      });
    }

    signoutHandler() {
      console.log('Signing out');
      firebase.auth().signOut();
      this.setState({ signedin: false });
    }

    render() {
      const {
        signedin,
      } = this.state;
      return (
        <App
          {...this.props}
          signoutHandler={this.signoutHandler}
          signedin={signedin}
        />
      );
    }
  };
}

export default withFirebase;
