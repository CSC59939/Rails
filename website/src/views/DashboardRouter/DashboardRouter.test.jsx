import React from 'react';
import renderer from 'react-test-renderer';
import {
  configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import firebase from 'firebase';
import { MemoryRouter } from 'react-router-dom';
import { DashboardRouter } from '..';

configure({ adapter: new Adapter() });


describe('DashboardRouter', () => {
  jest.spyOn(firebase, 'auth')
    .mockImplementation(() => ({
      currentUser: {
        displayName: 'testDisplayName',
        email: 'test@test.com',
        emailVerified: true,
        getIdToken: () => new Promise(((resolve) => {
          setTimeout(() => {
            resolve(1234);
          }, 0);
        })),
      },
    }));
  it('Render DashboardRouter without any eror', () => { // eslint-disable-line no-undef
    const component = renderer.create(
      <MemoryRouter>
        <DashboardRouter />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
