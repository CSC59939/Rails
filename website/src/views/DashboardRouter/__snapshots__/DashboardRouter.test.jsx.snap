// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRouter Render DashboardRouter without any eror 1`] = `
<div
  className="Container ant-layout"
  style={
    Object {
      "height": "100%",
    }
  }
>
  <div
    className="Header ant-layout-header"
  >
    <div
      className="ProfileInfo"
    >
      <a
        href="/dashboard"
      >
        <img
          alt="Rails"
          className="header-logo"
          src="logo.svg"
        />
      </a>
      <div
        className="rightSide"
      >
        <p
          className="name"
        >
          testDisplayName
        </p>
        <p
          className="email"
        >
          test@test.com
        </p>
      </div>
    </div>
    <div
      className="HeaderIcons"
    >
      <button
        className="IconButton"
        onClick={[Function]}
        type="button"
      >
        <i
          className="material-icons"
          style={null}
        >
          settings
        </i>
      </button>
      <a
        href="/signout"
        onClick={[Function]}
      >
        <button
          className="IconButton"
          onClick={[Function]}
          type="button"
        >
          <i
            className="material-icons"
            style={null}
          >
            exit_to_app
          </i>
        </button>
      </a>
    </div>
  </div>
  <div
    className="ant-layout"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div>
      <div />
    </div>
  </div>
</div>
`;
